swap_meet/vendor.py

```diff
@@ -1,2 +1,84 @@
+import uuid
+
+#from item import Item
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=[]):
+        self.inventory = inventory #each vendor will have an attribute named :inventory (an empty list)
+        
+    def add(self, item):
+        self.inventory.append(item)
+        print(self.inventory)
+        print(item)
+        return item 
+    
+    def remove(self, item): 
+        
+        if item not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item)
+            return item
+    
+    def get_by_id(self, id):
+        for item in self.inventory:
+            print(item.id)
+            print(id)
+            if item.id == id:
+                return item
+        return None
+        
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if not my_item in self.inventory or not their_item in other_vendor.inventory: 
+            return False
+        else:
+            self.remove(my_item)
+            self.add(their_item)
+            other_vendor.remove(their_item)
+            other_vendor.add(my_item)
+            return True 
+    # these functions will return uuid objects **(not int)**  
+
+    def swap_first_item(self, other_vendor):
+        if other_vendor.inventory == [] or self.inventory == []:
+            return False
+        else:
+            #look at the syntax here
+            other_vendor.add(self.inventory[0])
+            self.remove(self.inventory[0])
+            self.add(other_vendor.inventory[0])
+            other_vendor.remove(other_vendor.inventory[0])
+            return True
+    
+    def get_by_category(self, category):
+        objects = []
+
+        for item in self.inventory:
+            if item.get_category() == category:
+                objects.append(item)
+        return objects
```

Comment:
Notice this pattern of: 
```python
result_list = []
for element in source_list:
    if some_condition(element):
        result_list.append(element)
```

can be rewritten using a list comprehension as:
```python
result_list = [element for element in source_list if some_condition(element)]
```

Which here would look like:
```python
items_in_category = [item for item in self.inventory if item.get_category() == category]
return items_in_category
```

At first, this may seem more difficult to read, but comprehensions are a very common python style, so I encourage y’all to try working with them!